src/components/data/WdsTroubleshooter.js

```diff
@@ -0,0 +1,153 @@
+import _ from 'lodash/fp'
+import { Fragment, useState } from 'react'
+import { div, h, table, td, tr } from 'react-hyperscript-helpers'
+import { ClipboardButton } from 'src/components/ClipboardButton'
+import { ButtonPrimary } from 'src/components/common'
+import { icon } from 'src/components/icons'
+import Modal from 'src/components/Modal'
+import { Ajax } from 'src/libs/ajax'
+import { resolveWdsApp } from 'src/libs/ajax/data-table-providers/WdsDataTableProvider'
+import colors from 'src/libs/colors'
+import { useCancellation, useOnMount } from 'src/libs/react-utils'
+import * as Style from 'src/libs/style'
+import * as Utils from 'src/libs/utils'
+
+
+export const WdsTroubleshooter = ({ onDismiss, workspaceId, mrgId }) => {
+  const [leoOk, setLeoOk] = useState(null)
+  const [wdsResponsive, setWdsResponsive] = useState(null)
+  const [version, setVersion] = useState(null)
+  const [wdsStatus, setWdsStatus] = useState(null)
+  const [wdsDbStatus, setWdsDbStatus] = useState(null)
+  const [wdsPingStatus, setWdsPingStatus] = useState(null)
+  const [appFound, setAppFound] = useState(null)
+  const [appRunning, setAppRunning] = useState(null)
+  const [proxyUrl, setProxyUrl] = useState(null)
+  const [defaultInstanceExists, setDefaultInstanceExists] = useState(null)
+
+  const signal = useCancellation()
+
+  useOnMount(() => {
+    Ajax(signal).Apps.getV2AppInfo(workspaceId).then(res => {
+      setLeoOk(res)
+      const foundApp = resolveWdsApp(res)
+      setAppFound(foundApp?.appName)
+      setAppRunning(foundApp?.status)
+      setProxyUrl(foundApp?.proxyUrls?.wds)
+      Ajax(signal).WorkspaceData.getVersion(foundApp?.proxyUrls?.wds).then(res => {
+        setWdsResponsive(true)
+        setVersion(res.git?.commit?.id)
+      }).catch(_ => {
+        setWdsResponsive(false)
+        setVersion('unknown')
+      })
+      Ajax(signal).WorkspaceData.getStatus(foundApp?.proxyUrls?.wds).then(res => {
+        setWdsStatus(res.status)
+        setWdsDbStatus(res.components?.db?.status)
+        setWdsPingStatus(res.components?.ping?.status)
+      }).catch(_ => {
+        setWdsStatus('unresponsive')
+        setWdsDbStatus('unknown')
+        setWdsPingStatus('unknown')
+      })
+      Ajax(signal).WorkspaceData.listInstances(foundApp?.proxyUrls?.wds).then(res => {
+        setDefaultInstanceExists(res.includes(workspaceId))
+      }).catch(_ => {
+        setDefaultInstanceExists('unknown')
+      })
+    }).catch(_ => {
+      setLeoOk([])
+      setAppFound('unknown')
+      setAppRunning('unknown')
+      setProxyUrl('unknown')
+      setWdsResponsive('unknown')
+      setVersion('unknown')
+      setWdsStatus('unresponsive')
+      setWdsDbStatus('unknown')
+      setWdsPingStatus('unknown')
```

Comment:
need to add `setDefaultInstanceExists` in this catch, else if Leo isn't responding the "Default Instance exists" row in the troubleshooter will spin infinitely